### PR TITLE
Help text tweaks

### PR DESCRIPTION
**Pull request by BradleyChatha**
I'll start updating the README PR at another point in time (probably after I get most of the other feature suggestions done).

# Command help text with every current feature.

JCLI will auto wrap text, so the Lorem Ipsum snippet was just to show what it looks like.

```bash
Usage: tool.exe complex <file> <output> --iterations|-i --config|-c [--verbose|-v]

Description:
    This is a super complex command.

Positional Args:
    file                         - A file to do... stuff with.

Named Args:
    --iterations,-i              - How many iterations of a thing to do.
    --verbose,-v                 - Toggle verbose output.

IO Flags:
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
    aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur si
    nt occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

    output                       - The output file path.
    --config,-c                  - Path to the config file.
```

# No command given. (Same as commandless `--help` except it also has an error.)

```bash
tool.exe: No command was given.
Available commands:
    complex                      - This is a super complex command.
    sub one                      - This is the first sub command.
    sub two                      - This is the other sub command.
```

# Unknown command.

```bash
tool.exe: Unknown command 'bad'.
Did you mean:
    complex                      - This is a super complex command.
    sub one                      - This is the first sub command.
    sub two                      - This is the other sub command.
```

# Unknown command (partial match). (Same as partial match `--help` except is also has an error.)

```bash
tool.exe: Unknown command 'sub'.
Did you mean:
    sub one                      - This is the first sub command.
    sub two                      - This is the other sub command.
```

# Too many positional args

```bash
Running .\tool.exe complex a b c
tool.exe: too many arguments starting at 'c'
```

# Missing required args

```bash
Running .\tool.exe complex
tool.exe: Missing required arguments --iterations, --config, <file>, <output>
```

# Unknown named arg

```bash
Running .\tool.exe complex --nan
tool.exe: Unknown named argument: 'nan'
```

# Invalid conversion

**This needs to be changed** but it'd probably be tied into #12 rather than this PR.

```bash
Running .\tool.exe complex a a --iterations number

std.conv.ConvException@G:\None-Game\D\tools\dmd2\windows\bin\..\..\src\phobos\std\conv.d(2378): Unexpected 'n' when converting from type string to type int
```

# Command throws uncaught exception

Stack traces are only displayed in debug builds.

```
Running .\tool.exe sub one
tool.exe: This is an error message.

STACK TRACE:
0x00007FF680A583A0 in app.SubCommand1.onExecute at G:\None-Game\D\tools\dmd2\windows\bin\..\..\src\druntime\import\object.d(2058)
...
```

# Validation failed

```
Running .\tool.exe sub two a
tool.exe: For positional arg 0(input): Validation error message here
```

```
Running .\tool.exe sub two --input a
tool.exe: For named argument input: Validation error message here
```

# Conclusion

I think those are all the unique states JCLI will display messages? Probably missed a few.

### CODE REVIEW

File: source/jaster/cli/core.d [hunk 1]
```diff
@@ -421,7 +432,7 @@ final class CommandLineInterface(Modules...)
                 else if(this._defaultCommand == CommandInfo.init)
                 {
                     parseResult.type      = ParseResultType.commandNotFound;
-                    parseResult.helpText ~= format("ERROR: Unknown command '%s'.\n\n", parseResult.argParserBeforeAttempt.front.value);
+                    parseResult.helpText ~= this.makeErrorf("Unknown command '%s'.\n", parseResult.argParserBeforeAttempt.front.value);
                     parseResult.helpText ~= this.createAvailableCommandsHelpText(parseResult.argParserBeforeAttempt).toString();
```
Comment (andrey-zherikov) [inline source/jaster/cli/core.d:436]: I believe that all help texts can be compile-time constants or be printed without memory allocation.

Comment (BradleyChatha) [reply source/jaster/cli/core.d:481]: Fortunately the minor amount of allocations we're doing here isn't worth stressing about. Going with `.format` and `.makeErrorf` everywhere isn't needed, but it's just convenient.

Comment (andrey-zherikov) [reply source/jaster/cli/core.d:481]: Yeah, this is not something that should be worried about.

**Comment by BradleyChatha** [comment]
@andrey-zherikov This is good to go if you're happy with these changes.

Things like #9 will be done in a different PR.

**Comment by andrey-zherikov** [comment]
Please feel free to land it. I'm not so familiar with the code yet.

**Pull request merged**